src/main/java/org/candlepin/model/Product.java

```diff
@@ -1340,8 +1350,12 @@ public int getEntityVersion() {
                     .sorted()
                     .forEach(builder::append);
             }
+            else {
+                builder.append((Object) null);
+            }
 
-            if (!this.branding.isEmpty()) {
+            Collection<Branding> branding = this.getBranding();
+            if (branding != null && !branding.isEmpty()) {
```

Comment:
It would be nice if the builder could handle these cases for us. There is a lot of repetition that could be just one of two methods on the builder.

Reply:
I had considered it, and still might, but with the ordering bits in the middle throwing a wrench in it, I'm not sure how I want the interface to look for it. And given it's only used in one place today, it wasn't a priority of mine. 

If I can come up with something that won't look wonky or be highly specific to what's already in place, I'll add something for this.

Reply:
I tried adding something here, but nothing worked that removed enough from the `Product.getEntityVersion` method. In a generic method, we can't assume the caller wants it sorted or filtered, and we don't know the mapping. So all we end up doing is abstracting out the lines that would add the prefix gate and the `.forEach` line. The rest of the logic ends up remaining in product. But I have reworked it to be more flow-through and consistent.